pkgs/development/ocaml-modules/ocamlformat/generic.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, fetchurl, ocaml-ng, version }:
+
+# The ocamlformat package have been split into two in version 0.25.1:
+# one for the library and one for the executable.
+# Both have the same sources and very similar dependencies.
+
+with ocaml-ng.ocamlPackages;
```

Comment:
why is this using `ocaml-ng.ocamlPackages`?

Doesn't this mean OCamlformat will always build against 4.14 libraries (the current ocamlPackages) even if you want `ocaml-ng.ocamlPackages_4_12` or something?

Reply:
OCamlformat is not in `ocamlPackages`, it's at the root of nixpkgs. It has to build against the "main" version of ocaml (the one exposed in the root of nixpkgs).

Reply:
This is a good point though, because while ocamlformat is a tool and the version of OCaml it was build with doesn't matter, the new library should be part of ocamlPackages, just like ocamlformat-rpc-lib is.

Reply:
It would make sense indeed. That wouldn't be the first package defined in `ocamlPackages` and re-exported on all-packages, `reason` is like this too:

```
  inherit (ocamlPackages) reason;
```

I can work on this.

Reply:
I opened a PR: https://github.com/NixOS/nixpkgs/pull/241708